app/models/service_template.rb

```diff
@@ -103,12 +88,26 @@ def self.with_additional_tenants
     references(table_name, :tenants).includes(:service_template_tenants => :tenant)
   end
 
+  def self.all_catalog_item_types
+    @all_catalog_item_types ||= begin
+      builtin_catalog_item_types = {
+        "generic"               => N_("Generic"),
+        "generic_orchestration" => N_("Orchestration"),
+      }
+
+      ExtManagementSystem.supported_types_for_catalog
+        .flat_map(&:catalog_types)
+        .reduce(builtin_catalog_item_types, :merge)
+    end
+  end
+
   def self.catalog_item_types
-    ci_types = Set.new(Rbac.filtered(ExtManagementSystem.all).flat_map(&:supported_catalog_types))
+    ci_types = Set.new(Rbac.filtered(ExtManagementSystem.all).flat_map(&:catalog_types).reduce({}, :merge).keys)
```

Comment:
This should have a supported check somewhere in here, right?

Reply:
Though I guess the supported check below would suffice...just feels like it's ripe for a mistake if someone changes the below.

Reply:
:+1: added a select on `:supported_for_catalog?` here